### PR TITLE
Fix git submodules path issue

### LINKED ISSUE #75608: Undefined behaviour when using submodules and '.git' is present anywhere in the repo path
### Summary

See: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/git.py#L760 :

```py
            repo_path = os.path.join(repo_path.split('.git')[0], gitdir)
```

the intention of this code seems to be to strip `.git` from the end of the path.

If the repo path contains `.git` anywhere, like `/tmp/music.gittern/myrepo`, tasks will unexpectedly fail:

```
fatal: [localhost]: FAILED! => {"changed": false, "details": "/tmp/music/../.git/modules/submodule is not a directory", "msg": "Current repo does not have a valid reference to a separate Git dir or it refers to the invalid path"}
```

For a playbook like:
```yaml
---
- hosts: localhost
  tasks:
   - name: test git
     git: 
        repo: ...
        dest: /tmp/music.gittern/myrepo/submodule
```

I can't tell if this is a potential security issue or just an unexpected corner case. What I am pretty sure I can tell, is that that's not how one strips `.git` from the end of a path.

### Issue Type

Bug Report

### Component Name

git

### Ansible Version

```console
$ ansible --version
2.11
```

### Configuration

```console
$ ansible-config dump --only-changed
ANSIBLE_NOCOWS(/home/enrico/dev/transilience/ansible.cfg) = True
```

### OS / Environment

Debian Bullseye

### Steps to Reproduce

Use submodules in a repo whose path contains `.git` anywhere as part of a file name.
For example:
```yaml
---
- hosts: localhost
  tasks:
   - name: test git
     git: 
        repo: ...
        dest: /tmp/music.gittern/myrepo/submodule
```

### Expected Results

I'd expect directory names like `gato.gito` not to be interpreted specially by mistake

### Actual Results

```console
fatal: [localhost]: FAILED! => {"changed": false, "details": "/tmp/music/../.git/modules/submodule is not a directory", "msg": "Current repo does not have a valid reference to a separate Git dir or it refers to the invalid path"}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by maulikd**
##### SUMMARY
Fixes #75608

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
 `lib/ansible/modules/git.py`

### CODE REVIEW

File: test/integration/targets/git/tasks/submodules.yml [hunk 1]
```diff
@@ -122,3 +122,29 @@
...
+- name: SUBMODULES | Test that cloning submodule with .git in directory name works
+  git:
+    repo: "{{ repo_submodule1 }}"
+    dest: "{{ checkout_dir }}/test.gitdir/submodule1"
+
+- name: SUBMODULES | List submodule1
+  command: 'ls -1a {{ checkout_dir }}/test.gitdir/submodule1'
+  register: submodule1
+
+- name: SUBMODULES | Ensure submodule1 is at the appropriate commit
```
Comment (s-hertel) [inline test/integration/targets/git/tasks/submodules.yml:148]: These tests also pass for me on devel. There should be a test that reproduces the bug in #75608 and only passes with the patch.

**Review by s-hertel — DISMISSED** [review]

File: test/integration/targets/git/tasks/submodules.yml (hunk 1 above)
Comment (maulikd) [reply test/integration/targets/git/tasks/submodules.yml:148]: This should be fixed now.

**Review by s-hertel — APPROVED** [review]
LGTM

**Review by nitzmahone — APPROVED** [review]

**Comment by s-hertel** [comment]
@maulikd Thanks for the fixing this! Awesome first contribution.
